src/locate/synctex.ts

```diff
@@ -247,6 +319,65 @@ function callSyncTeXToPDF(line: number, col: number, filePath: string, pdfFile:
     })
 }
 
+/**
+ * Call SyncTeX to PDF for a specific line and character position.
+ *
+ * This function calls the SyncTeX binary to retrieve PDF information for a
+ * given line as a range in a TeX file. It returns a promise
+ * resolving to a SyncTeXRecordToPDFAllList object.
+ *
+ * @param line - The line number in the TeX file.
+ * @param filePath - The path of the TeX file.
+ * @param pdfFile - The path of the PDF file.
+ * @returns A promise resolving to a SyncTeXRecordToPDFAllList object.
+ */
+function callSyncTeXToPDFRange(line: number, filePath: string, pdfFile: string): Thenable<SyncTeXRecordToPDFAllList> {
```

Comment:
This function is largely similar to the existing `callSyncTeXToPDF`. Please instead merge the two functions and use a toggle parameter to distinguish the logic differences.